control/iosys.py

```diff
@@ -928,18 +930,21 @@ def __init__(self, syslist, connections=[], inplist=[], outlist=[],
             sysobj_list.append(sys)
             sysname_list.append(sys.name)
 
+        # This might cause a problem if there are duplicate names...
+        states = states or nstatenames
```

Comment:
What about if `states == 0` is an empty statenames list a valid parameter for the constructor below?

Reply:
nstatenames was the wrong way to do things. I fixed this differently in the new code, so this is not an issue anymore.